metals/src/main/scala/scala/meta/internal/metals/BloopServers.scala

```diff
@@ -304,31 +287,26 @@ final class BloopServers(
       maybeRequestedMetalsJavaHome: Option[String],
       reconnect: () => Future[BuildChange]
   ): Future[Unit] = {
-
     val result =
-      for { // if  bloopGlobalJsonFilePath is defined
-        bloopGlobalJsonFilePath <- getBloopFilePath(fileName = "bloop.json")
-        bloopCreatedByMetalsFilePath <- getBloopFilePath(fileName =
-          "created_by_metals.lock"
-        )
+      for {
+        bloopPath <- bloopJsonPath
         (maybeBloopGlobalJsonJavaHome, maybeBloopGlobalJsonJvmProperties) =
-          maybeLoadBloopGlobalJsonFile(bloopGlobalJsonFilePath)
-
+          maybeLoadBloopGlobalJsonFile(bloopPath)
         bloopJsonUpdateCause <-
           if (
-            maybeRequestedBloopJvmProperties != maybeBloopGlobalJsonJvmProperties
-            && maybeRequestedBloopJvmProperties.nonEmpty
+            maybeRequestedBloopJvmProperties
+              .exists(requested =>
+                requested != maybeBloopGlobalJsonJvmProperties
+              )
           ) Some(BloopJsonUpdateCause.JVM_OPTS)
           else if (maybeRequestedMetalsJavaHome != maybeBloopGlobalJsonJavaHome)
             Some(BloopJsonUpdateCause.JAVA_HOME)
           else None
-        maybeBloopJvmProperties = maybeRequestedBloopJvmProperties.orElse(
+        maybeBloopJvmProperties = maybeRequestedBloopJvmProperties.getOrElse(
           maybeBloopGlobalJsonJvmProperties
```

Comment:
so `maybeBloopGlobalJsonJvmProperties` is always defined?

Reply:
Yes, if it's an empty list it results in no options set for Bloop

Reply:
So it covers also the scenario when we want to remove the properties, right?

Reply:
Yes, that works